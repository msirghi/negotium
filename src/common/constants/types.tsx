import React from 'react';
import { OverridableComponent } from '@mui/types';
import { SvgIconTypeMap } from '@mui/material';

export type MainMenuItem = {
  title: string;
  Icon: OverridableComponent<SvgIconTypeMap<{}, 'svg'>> & { muiName: string };
};
