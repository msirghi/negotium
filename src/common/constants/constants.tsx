import { MainMenuItem, ProjectColor, SiteTheme } from './types';
import StarBorderIcon from '@mui/icons-material/StarBorder';
import InboxIcon from '@mui/icons-material/Inbox';
import DateRangeIcon from '@mui/icons-material/DateRange';
import DoneIcon from '@mui/icons-material/Done';
import { SnackbarOrigin } from 'notistack';
import colors from '../styles/colors';
import { Descendant } from 'slate';

export const MAIN_MENU_ITEMS: MainMenuItem[] = [
  {
    Icon: InboxIcon,
    title: 'pageTitles.inbox',
    route: '/home/inbox',
  },
  {
    Icon: StarBorderIcon,
    title: 'pageTitles.today',
    route: '/home/today',
  },
  {
    Icon: DateRangeIcon,
    title: 'pageTitles.upcoming',
    route: '/home/upcoming',
  },
  // {
  //   Icon: DoneIcon,
  //   title: 'Completed',
  //   route: '/home/completed',
  // },
];

export const PROJECT_NAME = 'Negotium';

export const BASE_API_URL = 'http://localhost:4000';

export const BASE_API_URL_V1 = 'https://negotium-api.herokuapp.com/api/v1';

export const MAX_PROJECT_LIST_COUNT = 5;

export const SNACKBAR_POSITIONS: { [key: string]: SnackbarOrigin } = {
  BOTTOM_CENTER: { horizontal: 'center', vertical: 'bottom' },
};

export const PROJECT_COLORS: ProjectColor[] = [
  {
    color: 'rgb(219, 64, 53)',
    name: 'Red',
  },
  {
    color: 'rgb(255, 153, 51)',
    name: 'Orange',
  },

  {
    color: 'rgb(250, 208, 0)',
    name: 'Yellow',
  },
];

export const MENTION_ARRAY_KEYWORDS = ['Today', 'Tomorrow'];

export const SUPPORTED_LANGUAGES = [
  {
    code: 'en',
    title: 'English',
  },
  {
    code: 'ru',
    title: 'Russian',
  },
];

export const HOME_VIEW_LIST = [
  {
    title: 'Inbox',
    url: '/home/inbox',
  },
  {
    title: 'Today',
    url: '/home/today',
  },
];

export const TIME_FORMATS = [
  {
    title: '13:00',
    key: '24h',
  },
  {
    title: '1pm',
    key: 'am/pm',
  },
];

export const tempAccountInfo = {
  id: '1',
  email: 'mihail.sirghi@gmail.com',
  name: 'Mihail',
};

export const pagesWithoutWrapper = ['/login', '/registration'];

export const passwordStatusColors = {
  tooWeak: 'red',
  weak: 'orange',
  medium: 'purple',
  strong: 'green',
};

export const siteThemes: SiteTheme[] = [
  {
    color: colors.primaries.lightBlue_1,
    label: 'Default',
    internalKey: 'default',
  },
  {
    color: colors.black,
    label: 'Noir',
    internalKey: 'noir',
  },
];

export const initialRichTextValue: Descendant[] = [
  {
    type: 'paragraph',
    children: [{ text: '' }],
  },
];

export const ACTIVATION_CODE_VAR_NAME = 'ac';
