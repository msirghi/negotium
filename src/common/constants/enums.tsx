export enum PASSWORD_STRENGTH_STATUS {
  TOO_WEAK = 'Too weak',
  WEAK = 'Weak',
  MEDIUM = 'Medium',
  STRONG = 'Strong',
}

export enum ROW_DIRECTION {
  ROW = 'row',
  COLUMN = 'column',
}

export enum ACCOUNT_SETTINGS_CHANGE_MODE {
  EMAIL = 'EMAIL',
  PASSWORD = 'PASSWORD',
}

export enum HttpMethod {
  DELETE = 'DELETE',
  GET = 'GET',
  HEAD = 'HEAD',
  OPTIONS = 'OPTIONS',
  PATCH = 'PATCH',
  POST = 'POST',
  PUT = 'PUT',
}
