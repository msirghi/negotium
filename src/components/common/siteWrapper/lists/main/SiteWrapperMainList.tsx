import { List } from '@mui/material';
import ListItem from '@mui/material/ListItem';
import ListItemIcon from '@mui/material/ListItemIcon';
import ListItemText from '@mui/material/ListItemText';
import { MAIN_MENU_ITEMS } from '../../../../../common/constants/constants';
import { ListItemTitle } from '../listItemTitle/ListItemTitle';
import styles from './SiteWrapperMainList.module.scss';
import { useRouter } from 'next/router';
import { Box } from '@mui/system';
import colors from '../../../../../common/styles/colors';
import { makeStyles } from '@mui/styles';

const useStyles = makeStyles({
  activeItem: {
    backgroundColor: colors.primaries.lightBlue_1,
    color: colors.white,
    borderRadius: 10,
    transition: 'all .1s ease-in-out',
  },
  activeIcon: {
    color: colors.white,
  },
});

export const SiteWrapperMainList = () => {
  const router = useRouter();
  const classes = useStyles();

  const isActive = (route: string) => {
    return router.route.includes(route.toLowerCase());
  };

  const onItemClick = (route: string) => {
    router.push(route);
  };

  return (
    <List sx={{ padding: 1 }}>
      {MAIN_MENU_ITEMS.map(({ Icon, title, route }) => {
        const isItemActive = isActive(route);
        return (
          <Box
            key={title}
            className={isItemActive ? classes.activeItem : ''}
            sx={{ borderRadius: 15 }}
          >
            <ListItem button onClick={() => onItemClick(route)}>
              <ListItemIcon>
                <Icon
                  fontSize={'small'}
                  className={isItemActive ? classes.activeIcon : ''}
                />
              </ListItemIcon>
              <ListItemText primary={<ListItemTitle title={title} />} />
            </ListItem>
          </Box>
        );
      })}
    </List>
  );
};
