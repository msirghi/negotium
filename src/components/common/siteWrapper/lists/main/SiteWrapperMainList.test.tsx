import { SiteWrapperMainList } from './SiteWrapperMainList';
import renderer from 'react-test-renderer';
import { MAIN_MENU_ITEMS } from '../../../../../common/constants/constants';
import { mount } from 'enzyme';
import { ListItem } from '@mui/material';

describe('SiteWrapperMainList', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<SiteWrapperMainList />);
    expect(tree).toMatchSnapshot();
  });

  it(`should render ${MAIN_MENU_ITEMS.length} list item`, () => {
    const wrapper = mount(<SiteWrapperMainList />);
    expect(wrapper.find(ListItem)).toHaveLength(MAIN_MENU_ITEMS.length);
  });
});
