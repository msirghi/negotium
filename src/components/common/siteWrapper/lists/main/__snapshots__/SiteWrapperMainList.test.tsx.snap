// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteWrapperMainList should match snapshot 1`] = `
<ul
  className="MuiList-root MuiList-padding css-2x8mgg-MuiList-root"
>
  <div
    className="MuiBox-root css-arly7"
  >
    <div
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-5qtusp-MuiButtonBase-root-MuiListItem-root"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="InboxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H4.99c-1.11 0-1.98.89-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.11-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
          >
            Inbox
          </p>
        </span>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root css-arly7"
  >
    <div
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-5qtusp-MuiButtonBase-root-MuiListItem-root"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="StarBorderIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
          >
            Today
          </p>
        </span>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root css-arly7"
  >
    <div
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-5qtusp-MuiButtonBase-root-MuiListItem-root"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="DateRangeIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
          >
            Upcoming
          </p>
        </span>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root css-arly7"
  >
    <div
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-5qtusp-MuiButtonBase-root-MuiListItem-root"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="DoneIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
          >
            Completed
          </p>
        </span>
      </div>
    </div>
  </div>
</ul>
`;
